extensions/wikia/Search/classes/Hooks.php

```diff
@@ -3,161 +3,196 @@
+	 *
+	 * @var \Wikia\Search\MediaWikiService
 	 */
 	protected static $service;
-	
+
 	/**
 	 * Sends delete request to article if it gets deleted
-	 * @param WikiPage $article
-	 * @param User $user
+	 *
+	 * @param \WikiPage $article
+	 * @param \User $user
 	 * @param integer $reason
 	 * @param integer $id
+	 *
+	 * @return true
 	 */
 	public static function onArticleDeleteComplete( &$article, \User &$user, $reason, $id ) {
-		return (new Indexer)->deleteArticle( $id );
+		return ( new Indexer )->deleteArticle( $id );
 	}
-	
+
 	/**
 	 * Reindexes the page
-	 * @param WikiPage $article
-	 * @param User $user
+	 *
+	 * @param \WikiPage $article
+	 * @param \User $user
 	 * @param string $text
 	 * @param string $summary
 	 * @param bool $minoredit
 	 * @param bool $watchthis
 	 * @param string $sectionanchor
 	 * @param array $flags
-	 * @param Revision $revision
+	 * @param \Revision $revision
 	 * @param int $status
 	 * @param int $baseRevId
+	 *
+	 * @return true
 	 */
-	public static function onArticleSaveComplete( &$article, &$user, $text, $summary,
-	        $minoredit, $watchthis, $sectionanchor, &$flags, $revision, &$status, $baseRevId ) {
-		return (new Indexer)->reindexBatch( array( $article->getTitle()->getArticleID() ) );
+	public static function onArticleSaveComplete(
+		&$article,
+		&$user,
+		$text,
+		$summary,
+		$minoredit,
+		$watchthis,
+		$sectionanchor,
+		&$flags,
+		$revision,
+		&$status,
+		$baseRevId
+	) {
+		return ( new Indexer )->reindexBatch( [ $article->getTitle()->getArticleID() ] );
 	}
-	
+
 	/**
 	 * Reindexes page on undelete
-	 * @param Title $title
+	 *
+	 * @param \Title $title
 	 * @param int $create
+	 *
+	 * @return true
 	 */
 	public static function onArticleUndelete( $title, $create ) {
-		return (new Indexer)->reindexBatch( array( $title->getArticleID() ) );
+		return ( new Indexer )->reindexBatch( [ $title->getArticleID() ] );
 	}
-	
+
 	/**
 	 * Issues a reindex event or deletes all docs, depending on whether a wiki is being closed or reopened
-	 * @param  int    $city_public
-	 * @param  int    $city_id
+	 *
+	 * @param  int $city_public
+	 * @param  int $city_id
 	 * @param  string $reason
+	 *
 	 * @return bool
 	 */
 	public static function onWikiFactoryPublicStatusChange( &$city_public, &$city_id, $reason ) {
-		return ( $city_public < 1 ) 
-		    ? (new Indexer)->deleteWikiDocs( $city_id )
-		    : (new Indexer)->reindexWiki( $city_id );
+		return ( $city_public < 1 ) ? ( new Indexer )->deleteWikiDocs( $city_id ) :
+			( new Indexer )->reindexWiki( $city_id );
 	}
-	
+
 	/**
-	 * Used to configure the user preference pane settings for search. 
+	 * Used to configure the user preference pane settings for search.
 	 * This is a registered hook function of the samme name.
-	 * 
-	 * @param User $user
+	 *
+	 * @param \User $user
 	 * @param array $defaultPreferences
-	 */ 
+	 */
 	public static function onGetPreferences( $user, &$defaultPreferences ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameter "$user". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)